File: docs/guides/csharp/src/searchInReplicaIndex.cs
namespace Algolia;

using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text.Json;
using Algolia.Search.Clients;
using Algolia.Search.Http;
using Algolia.Search.Models.Search;

class SearchInReplicaIndex
{
  async Task Main(string[] args)
  {
    try
    {
      var client = new SearchClient(new SearchConfig("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY"));

      var query = "query";

      // 1. Change the sort dynamically based on the UI events
      var sortByPrice = false;

      // 2. Get the index name based on sortByPrice
      var indexName = sortByPrice ? "products_price_desc" : "products";

      // 3. Search on dynamic index name (primary or replica)
      await client.SearchSingleIndexAsync<Hit>(
        indexName,
        new SearchParams(new SearchParamsObject { Query = "query" })
      );
    }
    catch (Exception e)
    {
      Console.WriteLine(e.Message);
    }
  }
}

File: docs/guides/go/src/searchInReplicaIndex.go
package main

import (
	"github.com/algolia/algoliasearch-client-go/v4/algolia/search"
)

func searchInReplicaIndex() {
	client, err := search.NewClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")
	if err != nil {
		// The client can fail to initialize if you pass an invalid parameter.
		panic(err)
	}

	// 1. Change the sort dynamically based on the UI events
	sortByPrice := false

	// 2. Get the index name based on sortByPrice
	indexName := "products"
	if sortByPrice {
		indexName = "products_price_desc"
	}

	// 3. Search on dynamic index name (primary or replica)
	_, err = client.SearchSingleIndex(client.NewApiSearchSingleIndexRequest(
		indexName).WithSearchParams(search.SearchParamsObjectAsSearchParams(
		search.NewEmptySearchParamsObject().SetQuery("query"))))
	if err != nil {
		panic(err)
	}
}
